### ISSUE TITLE
PHP 7.4 Incompatibility of paypal/module-braintree-core 4.4.0-p2

### ISSUE THREAD

**Issue by norgeindian** [magento/magento2#38150]
### Preconditions and environment

- Magento version 2.4.5.-p2

### Steps to reproduce

1. Use PHP 7.4
2. Run `composer require paypal/module-braintree-core:4.4.0-p2`
3. According to the `composer.json`, this should be compatible with PHP 7.4 (`"php": "~7.4.0||~8.1.0",`)
4. Run `bin/magento setup:di:compile`

### Expected result

`bin/magento setup:di:compile` should not throw any error

### Actual result

`bin/magento setup:di:compile` shows the following error message: `MiBsyntax error, unexpected '|', expecting ';' or '`

### Additional information

Checking that issue more detailed, you will come to the class `\PayPal\Braintree\Gateway\Command\GetPaymentNonceCommand` with the following method:
```
public function execute(array $commandSubject): ArrayResult|ResultInterface|null
    {
        $publicHash = $this->subjectReader->readPublicHash($commandSubject);
...
```
Unfortunately, union types for return types are a PHP 8 feature and not available in PHP 7.4
So in end it seems as if this version of the module is not compatible with PHP 7.4, even though it is stated.

### Triage and priority

- [X] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @norgeindian,

Thank you for reporting and collaboration.

As per the Magento system requirements https://experienceleague.adobe.com/docs/commerce-operations/installation-guide/system-requirements.html Magento 2.4.5-p2 supports with php 8.1.


Could you please check system requirements and please let us know if you are facing any issue.

Thanks.

**Comment by norgeindian** [comment]
@engcom-Bravo , that is correct. Magento indeed officially only supports PHP 8, even if it still works with PHP 7.4.
But `paypal/module-braintree-core:4.4.0-p2` states to be compatible with PHP 7.4, which seems to be incorrect as I described.

**Comment by hostep** [comment]
Was reported before in: https://github.com/magento/magento2/issues/37935

And even before that, I alerted at least 2 people from Adobe via Slack around this I believe somewhere in August 2023, they promised me they would deliver the message to the paypal team, but it seems like nothing has happened so far. And my guess would be that this will never get fixed. So I think your only chance is to replace the PHP 7.4 incompatible code with a composer patch.

It's also becoming time to move on to PHP 8, most 3rd party modules are compatible with 8.1 and even 8.2 by now, we hardly run into problems with PHP compatibility these days anymore. It's just your own custom code that you'll need to update if it's not compatible yet.

**Comment by engcom-Bravo** [comment]
@hostep Thanks for your contribution here.

Hi @norgeindian,

Thanks for your update.

Currently Magento supports with php 8.1 and 8.2 could you please check the issue in the Latest released Magento Version and Magenti 2.4-develop branch.

Thanks.

**Comment by engcom-Bravo** [comment]
Hi @norgeindian,

We have noticed that this issue has not been updated since long time. Hence we assume that this issue is fixed now, so we are closing it. Please feel to raise a fresh ticket or reopen this ticket if you need more assistance on this.

Thanks
